### DISCUSSION TITLE
An endpoint that returns bytes

### DISCUSSION THREAD

**Discussion by GTorreil** [vitalik/django-ninja#734] [Q&A]
Hello there :) 

I'm building an API that needs to distribute a firmware to update devices. I've set up an endpoint like so :

``` python
@router.get(
    "/latest/download/",
    response={
        200: bytes,
        404: MessageResponse,
    },
    summary="Télécharger le dernier firmware",
)
def download_latest(request):
    """
    Cet endpoint permet de télécharger le dernier firmware disponible.
    """
    firmware = get_latest_firmware()

    if firmware is None:
        return 404, MessageResponse(message="Aucun firmware disponible.")

    return firmware.bin.read()
```

I've also created a renderer for this kind of response : 

``` python
class BytesRenderer(BaseRenderer):
    """
    Un renderer qui retourne les octets bruts.
    """

    media_type = "application/octet-stream"

    def render(self, request: HttpRequest, data: Any, *, response_status: int) -> Any:
        return data
```

Now i'd like to use this renderer on the `download_latest` endpoint, but I want to keep using the JSON renderer for my other endpoints. I haven't found a `renderer` argument on the view decorator, am I missing something ? 

How can I return an octet stream with this endpoint ? 

Thanks a lot,

**Comment by vitalik** [comment]
@GTorreil  Just use the httpresponse ?

```Python
from django.http import HttpResponse

def download_latest(request):
    ...
    return HttpResponse(firmware.bin.read())
```

**Reply by GTorreil** [reply]
Oooh, I missed that, indeed. Thanks a lot ! :)

For future reference : https://django-ninja.rest-framework.com/guides/response/?h=httprespons#django-http-responses
